### PR TITLE
Unpin more dependencies in package.json

### PR DESCRIPTION
**Pull request by scinos**
#### Changes proposed in this Pull Request

Unpin a few dependencies in package.json (similar to #41191)

#### Testing instructions

* Verify there are no packages removed/added in yarn.lock

* Validate the dependency tree has not changed:
    * Install dependencies and generate the tree for this branch: `find . -type d -name node_modules | xargs rm -fr && yarn && NODE_OPTIONS=--max_old_space_size=8192 yarn run effective-module-tree -o list > ../branch-tree`
    * Install dependencies and generate the tree for master (tested with 8f2c0aa): `find . -type d -name node_modules | xargs rm -fr && yarn && NODE_OPTIONS=--max_old_space_size=8192 yarn run effective-module-tree -o list > ../master-tree`
    * Diff both trees and check there are no differences: `diff ../branch-tree ../master-tree`

### CODE REVIEW

**Review by sarayourfriend — APPROVED** [review]
I do see differences now when I diff with up-to-date master, however I'm 93.56% sure that they make sense given the deps you've unpinned. Might be worth taking a quick look before merging just to have another set of eyes on it. Otherwise LGTM, loving this clean up stuff 😁

**Comment by scinos** [comment]
@saramarcondes Looks like there were changes in maser that I had not rebase into this branch. 
With the current master (8f2c0aa896), I get no differences:

```diff
2c2
< $ /Users/sergio/src/automattic/wp-calypso2/node_modules/.bin/effective-module-tree -o list
---
> $ /Users/sergio/src/automattic/wp-calypso/node_modules/.bin/effective-module-tree -o list
3063878c3063878
< Done in 57.19s.
---
> Done in 28.07s.
```

**Pull request merged**
